pkg/tfgen/generate.go

```diff
@@ -549,6 +564,15 @@ func (t *propertyType) equals(other *propertyType) bool {
 	if len(t.properties) != len(other.properties) {
 		return false
 	}
+	switch {
+	case t.typePrefixOverride != nil && other.typePrefixOverride == nil:
+		return false
+	case t.typePrefixOverride == nil && other.typePrefixOverride != nil:
+		return false
+	case t.typePrefixOverride != nil && other.typePrefixOverride != nil &&
+		*t.typePrefixOverride != *other.typePrefixOverride:
+		return false
+	}
```

Comment:
Nit:

I think this is clearer as an `if` statement. It looks odd to me to have 3 branches that all lead to the same body.

```go
	if (t.typePrefixOverride != nil && other.typePrefixOverride == nil) ||
		(t.typePrefixOverride == nil && other.typePrefixOverride != nil) ||
		(t.typePrefixOverride != nil && other.typePrefixOverride != nil &&
			*t.typePrefixOverride != *other.typePrefixOverride) {
		return false
	}
```

That said, this is just an inequality check made painful by nil values. Nitpicking, I would express this as the inverse of an equality check:

```suggestion
if eq := t.typePrefixOverride == other.typePrefixOverride || 
	(t.typePrefixOverride != nil && other.typePrefixOverride != nil &&
		*t.typePrefixOverride == *other.typePrefixOverride); !eq {
	return false
}
```

Reply:
This is an F# habit  that may be out of place but I find individual cases more tractable than a large bool expr.